Terminal.Gui/Core/Application.cs

```diff
@@ -682,21 +682,41 @@ public static void Run<T> () where T : Toplevel, new()
 		///     the <see cref="RunLoop(RunState, bool)"/> method will only process any pending events, timers, idle handlers and
 		///     then return control immediately.
 		///   </para>
+		///   <para>
+		///     When <paramref name="errorHandler"/> is null the exception is rethrown, when it returns true the application is resumed and when false method exits gracefully.
+		///   </para>
 		/// </remarks>
 		/// <param name="view">The <see cref="Toplevel"/> tu run modally.</param>
-		public static void Run (Toplevel view)
+		/// <param name="errorHandler">Handler for any unhandled exceptions (resumes when returns true, rethrows when null).</param>
+		public static void Run (Toplevel view, Func<Exception, bool> errorHandler = null)
 		{
-			var runToken = Begin (view);
-			RunLoop (runToken);
-			End (runToken);
+			while (true)
+			{
+				var resume = false;
+				try
+				{
+					var runToken = Begin (view);
+					RunLoop (runToken);
+					End (runToken);
+				} 
+				catch (Exception error) 
+				{
+					if (errorHandler == null)
+						throw;
+					resume = errorHandler (error);
+				}
+				if (resume)
```

Comment:
Very debatable, but I think this code would be more clear if it was

```
var resume = true;
while (resume)
{
   try
   {
       resume = false;
       ...
   }
   catch
   {
       if (errorHandler == null)
       {
             throw;
       }
       resume = errorHandler(error);
   }
}
```

At the very least, please put brackets around the `throw;` per our style guidelines (always use enclosing brackets for `if`).

Reply:
Done, thanks for feedback.